### PR TITLE
SVSM Mode fixes

### PR DESCRIPTION
**Pull request by osteffenrh**
Here are some fixes and improvements for the SVSM mode:
- Check that the OVMF_VARS size is supplied when `--mode snp:svsm` is used
- Add the option to specify the OVMF_VARS file directly, instead of it's size. Adding the `--vars-file` option.
- [ToDo]: Update README, add an example for SVSM mode

### CODE REVIEW

File: sevsnpmeasure/cli.py [hunk 1]
```diff
@@ -98,13 +108,24 @@ def main() -> int:
...
 
+        if sev_mode == SevMode.SEV_SNP_SVSM:
+
+            if args.vars_file is not None:
+                vars_size = pathlib.Path(args.vars_file).stat().st_size
+            else:
+                vars_size = args.vars_size
+
+            if vars_size is None:
+                parser.error("sev:svsm mode requires --vars-size to be set")
```
Comment (dubek) [inline sevsnpmeasure/cli.py:122]:
```suggestion
                parser.error("snp:svsm mode requires --vars-size to be set")
```

File: setup.cfg [hunk 2]
```diff
@@ -31,5 +31,5 @@ console_scripts =
     snp-create-id-block = sevsnpmeasure.id_block:main
 
 [flake8]
-max-complexity = 10
+max-complexity = 12
```
Comment (dubek) [inline setup.cfg:34]: Please also change `Makefile` accordingly (this is what breaks the CI in github).

**Review by dubek — CHANGES_REQUESTED** [review]
Thanks @osteffenrh . Small changes below.
Are you planning to add the README change as well?

File: setup.cfg (hunk 2 above)
Comment (osteffenrh) [reply setup.cfg:34]: Or remove the arguments from the call in Makefle and have it pick up the settings from setup.cfg?

**Comment by osteffenrh** [comment]
Thanks for taking a look already, @dubek.
I am not sure if I am happy with the command line options yet.
And the Readme changes I'll also add.

File: setup.cfg (hunk 2 above)
Comment (dubek) [reply setup.cfg:34]: Yes, you're right, if it automatically reads setup.cfg. Your way is even better.

File: sevsnpmeasure/cli.py [hunk 3]
```diff
@@ -102,14 +112,24 @@ def main() -> int:
...
         sev_mode = SevMode.from_str(args.mode)
 
+        if sev_mode == SevMode.SEV_SNP_SVSM:
+
+            if args.vars_file:
+                vars_size = pathlib.Path(args.vars_file).stat().st_size
+            elif args.vars_size:
+                vars_size = args.vars_size
+            else:
+                parser.error("snp:svsm mode requires --vars-size to be set")
```
Comment (dubek) [inline sevsnpmeasure/cli.py:125]: Maybe:
```suggestion
                parser.error("snp:svsm mode requires --vars-size or --vars-file")
```

File: README.md [hunk 4]
```diff
@@ -29,12 +29,15 @@ Clone the Github repo and run the script directly from the local directory:
...
-                       [--vars-size VARS_SIZE] [--svsm SVSM]
+usage: sev-snp-measure [-h] [--version] [-v] --mode
+                       {sev,seves,snp,snp:ovmf-hash,snp:svsm} [--vcpus N]
+                       [--vcpu-type CPUTYPE] [--vcpu-sig VALUE]
+                       [--vcpu-family FAMILY] [--vcpu-model MODEL]
+                       [--vcpu-stepping STEPPING] [--vmm-type VMMTYPE] --ovmf
+                       PATH [--kernel PATH] [--initrd PATH] [--append CMDLINE]
+                       [--guest-features VALUE] [--output-format {hex,base64}]
+                       [--snp-ovmf-hash HASH] [--dump-vmsa] [--svsm PATH]
+                       [--vars-size SIZE | --vars-file PATH]
```
Comment (dubek) [inline README.md:41]: nit: I use the following to add the usage message to the README:

```
COLUMNS=100 python3 ./sev-snp-measure.py --help
```

Hopefully this way you'll have more precise diffs.

(maybe I should document this somewhere or automate it in a Makefile task. hmm.)

**Review by dubek — CHANGES_REQUESTED** [review]
Looks good, thanks @osteffenrh for fixing and rebasing on top of the latest main. Few small documentation comments below, and then I can merge.

File: README.md (hunk 4 above)
Comment (osteffenrh) [reply README.md:41]: Ah, thanks for the hint!
I'll update it.

Comment (osteffenrh) [reply README.md]: `COLUMNS=100` :+1:

But I still get some unexpected changes.

**Review by dubek — APPROVED** [review]
LGTM. Thanks for the fixes.

File: README.md (hunk 4 above)
Comment (dubek) [reply README.md]: Looks OK to me.

**Review by fitzthum — APPROVED** [review]
LGTM

**Comment by dubek** [comment]
This is now part of the sev-snp-measure 0.0.9 release published to pypi. Thank you!

**Pull request merged**